cap/modules/search/sorter.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of CERN Analysis Preservation Framework.
+# Copyright (C) 2022 CERN.
+#
+# CERN Analysis Preservation Framework is free software; you can redistribute
+# it and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# CERN Analysis Preservation Framework is distributed in the hope that it will
+# be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with CERN Analysis Preservation Framework; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""CAP sorter factory for REST API."""
+
+import six
+from flask import current_app, request
+from invenio_records_rest.sorter import eval_field, parse_sort_field
+
+
+def cap_sorter_factory(search, index):
```

Comment:
we dont have to change the `default_sorter_factory`, just pass in `RECORDS_REST_SORT_OPTIONS` a callable field. like 

```
def example_sort(field_name, arguments=None):
    def inner(asc):
        return {field_name: {'order': 'asc' if asc else 'desc', 'unmapped_type': "date" }, }
    return inner

DEPOSIT_REST_SORT_OPTIONS.update(
    dict(
        deposits=dict(
            target_date=dict(
                title=_('Target Date (desc)'),
                fields=['initial.target_date'],
                unmapped_type="date",
                ignore_unmapped= False,
                missing="_last",
                default_order='desc',
                order=3,
            ),
            example=dict(
                title=_('Target Date'),
                fields=[example_sort('initial.target_date', {})],
                default_order='asc',
                order=4,
            )
        )
    )
)
```